modules/tensor_mechanics/src/materials/RankTwoTensorComponent.C

```diff
@@ -0,0 +1,60 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "RankTwoTensorComponent.h"
+#include "RankTwoScalarTools.h"
+
+registerMooseObject("TensorMechanicsApp", RankTwoTensorComponent);
+
+InputParameters
+RankTwoTensorComponent::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription("Access a component of a RankTwoTensor");
+  params.addRequiredParam<MaterialPropertyName>("rank_two_tensor",
+                                                "The rank two material tensor name");
+  params.addParam<std::string>("calculation_name",
```

Comment:
Change `calculation_name` to `property_name` and other related changes consistent with what I suggested for the cylindrical class.